autogen/oai/client.py

```diff
@@ -158,8 +123,19 @@ def _client(self, config: Dict[str, Any], openai_config: Dict[str, Any]) -> Open
         after removing extra kwargs.
         """
         openai_config = {**openai_config, **{k: v for k, v in config.items() if k in self.openai_kwargs}}
-        self._process_for_azure(openai_config, config)
-        client = OpenAI(**openai_config)
+        api_type = config.get("api_type")
+        if api_type is not None and api_type.startswith("azure"):
+            api_key = config.get("api_key", os.environ.get("AZURE_OPENAI_API_KEY"))
+            api_version = config.get("api_version", DEFAULT_AZURE_API_VERSION)
+            model = config.get("model")
+            base_url = config.get("base_url")
+            if base_url is None:
+                raise ValueError("to use azure openai api, base_url must be specified.")
```

Comment:
This part should be in `_process_for_azure`. Also, we need to consider the case where the dict is indeed specified with the right key name following the AzureOpenAI requirement.

Reply:
The idea is to remove `_process_for_azure()` for good and make all client preparation in `_client()`

Reply:
We currently use it for both client args and create args.